### PR TITLE
Websocket fix

### PR DESCRIPTION
**Pull request by saf6260**
#### Summary
This ticket addresses the heartbeat issues detailed in the [mattermost discussion forum](https://forum.mattermost.org/t/heartbeat-panic-issues-when-using-the-mattermost-sample-bot-and-mattermost-docker-preview-image/11215) for issues surrounding the direct use of a websocket client.

As per the comment, the code is unneeded and, therefore, removed to assist future developers for getting started with development of their own mattermost bot.

This pull request further updates the version of code for v1.15 and removes unneeded dependencies per code removal

#### Ticket Link
There is no ticket # for this assigned PR

### CODE REVIEW

File: bot_sample.go [hunk 1]
```diff
@@ -74,24 +72,6 @@ func main() {
...
-		for {
-			select {
-			case resp := <-webSocketClient.EventChannel:
-				HandleWebSocketResponse(resp)
-			case <-ticker.C:
-				webSocketClient.Conn.SetWriteDeadline(time.Now().Add(2 * time.Second))
-				if err := webSocketClient.Conn.WriteMessage(websocket.PingMessage, []byte{}); err != nil {
-					println("send heartbeat error: ", err)
-					return
-				}
```
Comment (agnivade) [inline bot_sample.go:90]: Hi @saf6260 ! It's only the ping part which is not required. The handling of the eventChannel can still be done.

Comment (saf6260) [reply bot_sample.go:90]: Ah, thank you for catching that! My mistake, the websocket event handler should be back

File: bot_sample.go [hunk 2]
```diff
@@ -74,20 +72,11 @@ func main() {
 
 	webSocketClient.Listen()
 
-	ticker := time.NewTicker(5 * time.Second)
-	defer ticker.Stop()
-
 	go func() {
 		for {
```
Comment (agnivade) [inline bot_sample.go:76]: We can further simplify this by just ranging the channel instead:

```go
for resp := range client.EventChannel {
  handle(resp)
}
```

**Review by agnivade — COMMENTED** [review]
Looking great now. Just a minor improvement suggestion.

File: bot_sample.go (hunk 2 above)
Comment (saf6260) [reply bot_sample.go]: Good catch! I updated the func() to use this syntax

**Review by agnivade — COMMENTED** [review]
Looks great. Requesting @streamer45 for a second look.

**Review by agnivade — APPROVED** [review]

**Review by streamer45 — APPROVED** [review]
Thank you @saf6260 :+1:

**Pull request merged**